test/ledger/grants.integration.test.js

```diff
@@ -0,0 +1,109 @@
+})
+
+test('ledger: grants: get known', async t => {
+  const personaId = v4().toLowerCase()
+
+  var response = await ledgerAgent.get('/v2/registrar/persona').expect(ok)
+  t.true(response.body.hasOwnProperty('registrarVK'))
+  const personaCredential = new anonize.Credential(personaId, response.body.registrarVK)
+
+  const keypair = tweetnacl.sign.keyPair()
+  console.log('created new ed25519 keypair')
+  console.log(JSON.stringify({
+    'publicKey': uint8tohex(keypair.publicKey),
+    'secretKey': uint8tohex(keypair.secretKey)
+  }))
+
+  const body = {
+    label: v4().toLowerCase(),
+    currency: 'BAT',
+    publicKey: uint8tohex(keypair.publicKey)
+  }
+  var octets = JSON.stringify(body)
+  var headers = {
+    digest: 'SHA-256=' + crypto.createHash('sha256').update(octets).digest('base64')
+  }
+
+  headers['signature'] = sign({
+    headers: headers,
+    keyId: 'primary',
+    secretKey: uint8tohex(keypair.secretKey)
+  }, { algorithm: 'ed25519' })
+
+  var payload = { requestType: 'httpSignature',
+    request: {
+      body: body,
+      headers: headers,
+      octets: octets
+    },
+    proof: personaCredential.request()
+  }
+  response = await ledgerAgent.post('/v2/registrar/persona/' + personaCredential.parameters.userId)
+    .send(payload).expect(ok)
+  t.true(response.body.hasOwnProperty('wallet'))
+  const paymentId = response.body.wallet.paymentId
+  t.true(response.body.wallet.hasOwnProperty('paymentId'))
+  t.true(response.body.wallet.hasOwnProperty('addresses'))
+  t.true(response.body.hasOwnProperty('verification'))
+
+  t.true(response.body.wallet.addresses.hasOwnProperty('BAT'))
+  t.true(response.body.wallet.addresses.hasOwnProperty('BTC'))
+  t.true(response.body.wallet.addresses.hasOwnProperty('CARD_ID'))
+  t.true(response.body.wallet.addresses.hasOwnProperty('ETH'))
+  t.true(response.body.wallet.addresses.hasOwnProperty('LTC'))
+
+  // get available grant
+  response = await ledgerAgent
+    .get('/v1/grants')
+    .expect(ok)
+
+  t.true(response.body.hasOwnProperty('promotionId'))
+
+  const promotionId = response.body.promotionId
+
+  // request grant
+  response = await ledgerAgent
+      .put(`/v1/grants/${paymentId}`)
+      .send({promotionId})
+      .expect(ok)
+  console.log(response.body)
+  t.true(response.body.hasOwnProperty('probi'))
+
+  const donateAmt = new BigNumber(response.body.probi).dividedBy('1e18').toNumber()
+  const desired = donateAmt.toString()
+
+  // try re-claiming grant, should return ok
+  response = await ledgerAgent
+      .put(`/v1/grants/${paymentId}`)
+      .send({'promotionId': promotionId})
+      .expect(ok)
+
+  do {
+    response = await ledgerAgent
+      .get(`/v2/wallet/${paymentId}?refresh=true&amount=${desired}&altcurrency=BAT`)
+    if (response.status === 503) await timeout(response.headers['retry-after'] * 1000)
+  } while (response.status === 503)
+  var err = ok(response)
+  if (err) throw err
+  t.false(response.body.hasOwnProperty('unsignedTx'))
```

Comment:
missing `unsignedTx` should be sufficient, but might be good to just toss in an explicit check that the wallet balance is zero